backend/fms_core/template_importer/row_handlers/library_qc/library_qc.py

```diff
@@ -139,6 +139,10 @@ def process_row_inner(self, sample_container, measures, process, process_measure
             update_qc_flags(sample=source_sample_obj,
                             quantity_flag=measures['quantity_flag'],
                             quality_flag=measures['quality_flag'])
+
+        # Add a warning if the sample has failed qc
+        if any([measures['quantity_flag'] == 'Failed', measures['quality_flag'] == 'Failed']):
+            self.warnings["flags"] = (f"Source library {source_sample_obj.name} has failed QC.")
```

Comment:
I would modify the message here to instead indicate that you are about to overwrite a failed QC flag.